presto-raptorx/src/main/java/com/facebook/presto/raptorx/systemtable/ChunkSystemTable.java

```diff
@@ -0,0 +1,141 @@
+
+    @Inject
+    public ChunkSystemTable(TransactionManager transactionManager, TypeManager typeManager)
+    {
+        this.transactionManager = requireNonNull(transactionManager, "transactionManager is null");
+
+        this.tableMetadata = new ConnectorTableMetadata(
+                new SchemaTableName("system", "chunks"),
+                ImmutableList.<ColumnMetadata>builder()
+                        .add(new ColumnMetadata("table_schema", VARCHAR))
+                        .add(new ColumnMetadata("table_name", VARCHAR))
+                        .add(new ColumnMetadata("chunk_id", BIGINT))
+                        .add(new ColumnMetadata("bucket_number", BIGINT))
+                        .add(new ColumnMetadata("uncompressed_size", BIGINT))
+                        .add(new ColumnMetadata("compressed_size", BIGINT))
+                        .add(new ColumnMetadata("row_count", BIGINT))
+                        .add(new ColumnMetadata("xxhash64", createVarcharType(16)))
+                        .build());
+    }
+
+    @Override
+    public Distribution getDistribution()
+    {
+        return SINGLE_COORDINATOR;
+    }
+
+    @Override
+    public ConnectorTableMetadata getTableMetadata()
+    {
+        return tableMetadata;
+    }
+
+    @Override
+    public ConnectorPageSource pageSource(ConnectorTransactionHandle transactionHandle, ConnectorSession session, TupleDomain<Integer> constraint)
+    {
+        Transaction transaction = transactionManager.get(transactionHandle);
+
+        Map<Integer, NullableValue> values = extractFixedValues(constraint).orElse(ImmutableMap.of());
+        Optional<String> schemaName = getStringValue(values.get(getColumnIndex(tableMetadata, "table_schema")));
+        Optional<String> tableName = getStringValue(values.get(getColumnIndex(tableMetadata, "table_name")));
+
+        PageListBuilder builder = PageListBuilder.forTable(tableMetadata);
+
+        Iterator<Page> iterator = listTables(transaction, schemaName, tableName).stream()
+                .map(table -> {
+                    builder.reset();
+                    writeTable(builder, transaction, table);
+                    return builder.build();
+                })
+                .flatMap(Collection::stream)
+                .iterator();
+
+        return new IteratorPageSource(iterator);
+    }
+
+    private static void writeTable(PageListBuilder builder, Transaction transaction, SchemaTableName tableName)
+    {
+        Optional<Long> tableId = transaction.getTableId(tableName);
+        if (!tableId.isPresent()) {
+            return;
+        }
+
+        Collection<ChunkMetadata> chunks = transaction.getChunks(tableId.get());
+        for (ChunkMetadata chunk : chunks) {
+            writeChunk(builder, tableName, chunk);
+        }
+    }
+
+    private static void writeChunk(PageListBuilder builder, SchemaTableName tableName, ChunkMetadata chunk)
+    {
+        builder.beginRow();
```

Comment:
move begin and and outside `writeChunk` so that it is consistent with other two system table impl?

Reply:
yeah